docs/latest/design/extending-envoy-gateway.md

```diff
@@ -149,9 +149,13 @@ latency between Envoy and the authentication service and also allows the end use
 Extending Envoy Gateway by using an external extension server which makes use of hook points in Envoy Gateway does come with some known issues that will have to be worked around.
 
 - Extension server handling its own state such as in the scenario where an extension establishes watches its own CRDs in the cluster can lead to a scenario where
-ensuring that the extension and Envoy Gateway are in-sync becomes a challenge.
-  - A potential workaround to this issue would be to have the extension use hook points in Envoy Gateway to replace Envoy Gateway's Kubernetes provider (the component that watches resources such as the 
-Gateway API CRDs) with a custom provider that can enable Envoy Gateway to be a single source of truth for configuration coming from the cluster.
+ensuring that the extension and Envoy Gateway are in-sync becomes a challenge. There are two potential solutions to this problem.
+  - Option 1: Create a new External Provider mechanism where an extension can take over all provider responsibility for Envoy Gateway. The current functionality of the provider would need to become a Library that EG would make use of 
```

Comment:
I think the option 1 suggested on the call today was - EG reconciles/watches only EG resources, and the extension watches the Extension resources

Reply:
I agree this is a better way to seperates concerns. However, if keeping EG and Extensions in sync poses a problem, I prefer 'the provider listens to all resources and passes them back to the extension as needed' approach. I think we should put too much responsibility on the Extension side.